### PR TITLE
ROX-9812: Provide Network Policy information from sensor to the policy matcher

### PR DESCRIPTION
**Pull request by fredrb**
## Description

Links the sensor detection of Network Policies with the boolean policy matches. This PR unlocks the possibility of triggering Network Policy `Missing (ingress|egress) Network Policy` violations, provided that the Network Policy information is provided in sensor. 

A few things to consider:
- ~~Currently, the Network Policies are only passed to deploytime detectors. This means that if Network Policy System Policy are also evaluated during runtime we might have false positives.~~ [It won't](https://github.com/stackrox/stackrox/blob/master/pkg/detection/runtime/detector.go#L99-L102)
- The Network Policy information is currently hardcoded as the consumption of Network Policies from k8s API is not fully implemented yet.  

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] ~~Evaluated and added CHANGELOG entry if required~~
- [ ] ~~Determined and documented upgrade steps~~
- [ ] ~~Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

No changes communicated to customers at this point

## Testing Performed

- Default policy tests for the new fields

### CODE REVIEW

**Review by lvalerom — APPROVED** [review]
LGTM

File: pkg/booleanpolicy/augmentedobjs/construct.go [hunk 1]
```diff
@@ -143,12 +143,19 @@ func ConstructDeploymentWithNetworkFlowInfo(
...
+func ConstructDeployment(deployment *storage.Deployment, images []*storage.Image, applied *NetworkPoliciesApplied) (*pathutil.AugmentedObj, error) {
 	obj := pathutil.NewAugmentedObj(deployment)
 	if len(images) != len(deployment.GetContainers()) {
 		return nil, errors.Errorf("deployment %s/%s had %d containers, but got %d images",
 			deployment.GetNamespace(), deployment.GetName(), len(deployment.GetContainers()), len(images))
 	}
+
+	appliedPolicies := pathutil.NewAugmentedObj(applied)
+	err := obj.AddAugmentedObjAt(appliedPolicies, pathutil.FieldStep(networkPoliciesAppliedKey))
+	if err != nil {
```
Comment (vikin91) [inline pkg/booleanpolicy/augmentedobjs/construct.go:155]: This could be one line

File: pkg/booleanpolicy/augmentedobjs/construct.go [hunk 2]
```diff
@@ -143,12 +143,19 @@ func ConstructDeploymentWithNetworkFlowInfo(
...
 // ConstructDeployment constructs the augmented deployment object.
-func ConstructDeployment(deployment *storage.Deployment, images []*storage.Image) (*pathutil.AugmentedObj, error) {
+func ConstructDeployment(deployment *storage.Deployment, images []*storage.Image, applied *NetworkPoliciesApplied) (*pathutil.AugmentedObj, error) {
 	obj := pathutil.NewAugmentedObj(deployment)
 	if len(images) != len(deployment.GetContainers()) {
 		return nil, errors.Errorf("deployment %s/%s had %d containers, but got %d images",
 			deployment.GetNamespace(), deployment.GetName(), len(deployment.GetContainers()), len(images))
 	}
+
+	appliedPolicies := pathutil.NewAugmentedObj(applied)
```
Comment (vikin91) [inline pkg/booleanpolicy/augmentedobjs/construct.go:153]: Is it okay passing `nil` to that function? It later does `reflect.ValueOf(nil)` which produces `<invalid reflect.Value>`

Try: `fmt.Printf("value: %+v", reflect.ValueOf(nil))`

File: pkg/booleanpolicy/default_policies_test.go [hunk 3]
```diff
@@ -3106,6 +3126,82 @@ func (suite *DefaultPoliciesTestSuite) TestLivenessProbePolicyCriteria() {
...
+	return violations
+}
+
+func (suite *DefaultPoliciesTestSuite) TestNetworkPolicyFields() {
+	if !features.NetworkPolicySystemPolicy.Enabled() {
+		return
+	}
+
+	testCases := map[string]struct {
+		netpolsApplied augmentedobjs.NetworkPoliciesApplied
```
Comment (vikin91) [inline pkg/booleanpolicy/default_policies_test.go:3144]: Could you please model this as pointer (`netpolsApplied *augmentedobjs.NetworkPoliciesApplied`) and add a test case with `nil`?

**Review by vikin91 — CHANGES_REQUESTED** [review]
It looks really good, but I just want to be sure that we are handling `nil`s correctly. Could you please add the test case for that (as in the comment)?

File: pkg/booleanpolicy/augmentedobjs/construct.go (hunk 2 above)
Comment (fredrb) [reply pkg/booleanpolicy/augmentedobjs/construct.go:153]: TIL that the reflect package treats `nil` values differently from `nil` pointers: https://go.dev/play/p/We8LUJzQBPh

Since we're providing a nil pointer to a struct the reflected value is `<nil>`. Added a test case to show that as well as you suggested below.

File: pkg/booleanpolicy/augmentedobjs/construct.go (hunk 1 above)
Comment (fredrb) [reply pkg/booleanpolicy/augmentedobjs/construct.go]: Done

File: pkg/booleanpolicy/evaluator/pathutil/augmented_obj.go [hunk 4]
```diff
@@ -77,6 +78,8 @@ type AugmentedObj struct {
 // paths within this object.
 func NewAugmentedObj(actualObj interface{}) *AugmentedObj {
 	value := reflect.ValueOf(actualObj)
+	strVal := fmt.Sprintf("%v\n", value)
+	fmt.Println(strVal)
```
Comment (vikin91) [inline pkg/booleanpolicy/evaluator/pathutil/augmented_obj.go:82]: Looks like leftover debugging

**Review by vikin91 — APPROVED** [review]
Thanks! 

I think there are some debugging leftovers - see comment

File: pkg/booleanpolicy/evaluator/pathutil/augmented_obj.go (hunk 4 above)
Comment (fredrb) [reply pkg/booleanpolicy/evaluator/pathutil/augmented_obj.go:82]: Oops
